invenio/modules/search/facet_builders.py

```diff
@@ -139,9 +141,15 @@ def get_facets_for_query(self, qid, limit=20, parent=None):
         """Return facet data."""
         from invenio.legacy.search_engine import get_most_popular_field_values,\
             get_field_tags
-        return get_most_popular_field_values(self.get_recids(qid),
-                                             get_field_tags(self.name)
-                                             )[0:limit]
+        most_popular_fields = get_most_popular_field_values(
+            self.get_recids(qid), get_field_tags(self.name))[0:limit]
+        most_popular_fields = map(lambda x: {
```

Comment:
- [ ] please use Python list comprehension (more effective)
- [ ] it would be nice to make the structure as similar to Elastic Search as possible where it makes sense (see http://www.elastic.co/guide/en/elasticsearch/reference/1.x/search-aggregations-bucket-nested-aggregation.html)